### ISSUE TITLE
3.10.1: 'python -m Crypto.SelfTest' OK, while 'python -m Crypto.SelfTest  --skip-slow-tests' fails...

### ISSUE THREAD

**Issue by Vangelis66** [Legrandin/pycryptodome#508]
OS: Windows Vista SP2 32-bit. updated fully to EoS
Python version: **3.7.9** (32-bit)
PyCryptodome version: **3.10.1**, installed from **PyPi** via 
`python -m pip install -U pycryptodome` => 

```
Requirement already satisfied: pycryptodome in c:\python379-32\lib\site-packages
 (3.9.9)
Collecting pycryptodome
  Downloading pycryptodome-3.10.1-cp35-abi3-win32.whl (1.6 MB)
     |████████████████████████████████| 1.6 MB 930 kB/s
Installing collected packages: pycryptodome
  Attempting uninstall: pycryptodome
    Found existing installation: pycryptodome 3.9.9
    Uninstalling pycryptodome-3.9.9:
      Successfully uninstalled pycryptodome-3.9.9
Successfully installed pycryptodome-3.10.1
```
As 3.10.1 **does not include test vectors**, I proceeded with: 
`python -m pip install -U pycryptodome-test-vectors` => 
```
Collecting pycryptodome_test_vectors
  Downloading https://files.pythonhosted.org/packages/fe/57/4969cd508aef58c3d272
0cf1cdee5f570124ea20ce52b8d83c1a8583e2dc/pycryptodome_test_vectors-1.0.2-py2.py3
-none-any.whl (11.9MB)
     |████████████████████████████████| 11.9MB 285kB/s
Installing collected packages: pycryptodome-test-vectors
Successfully installed pycryptodome-test-vectors-1.0.2
```
As suggested in the documentation, I then issued: 
`python -m Crypto.SelfTest` => 
```
kipping AESNI tests
Skipping test of PCLMULDQD in AES GCM
Skipping GMP tests on Windows
................................................................................
(snipped)
................................................................................
...................
----------------------------------------------------------------------
Ran 22979 tests in 1643.584s

OK
```
... so **no errors at all**; but when, out of curiosity, I issued: 

`python -m Crypto.SelfTest --skip-slow-tests` =>
```
Skipping slow tests
Skipping AESNI tests
Skipping test of PCLMULDQD in AES GCM
Skipping GMP tests on Windows
................................................................................
(snipped)
................................................................................
.......................E........................................................
..........................................................................
======================================================================
ERROR: runTest (Crypto.SelfTest.Signature.test_dss.TestVectorsECDSAWycheproof)
None
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python379-32\lib\site-packages\Crypto\SelfTest\Signature\test_dss.py"
, line 1050, in setUp
    self.add_tests("ecdsa_secp224r1_sha3_224_test.json")
  File "C:\Python379-32\lib\site-packages\Crypto\SelfTest\Signature\test_dss.py"
, line 1039, in add_tests
    'encoding': filter_encoding,
  File "C:\Python379-32\lib\site-packages\Crypto\SelfTest\loader.py", line 181,
in load_test_vectors_wycheproof
    common_group[k] = v(group)
  File "C:\Python379-32\lib\site-packages\Crypto\SelfTest\Signature\test_dss.py"
, line 1023, in filter_sha
    return get_hash_module(group['sha'])
  File "C:\Python379-32\lib\site-packages\Crypto\SelfTest\Signature\test_dss.py"
, line 937, in get_hash_module
    raise ValueError("Unknown hash algorithm: " + hash_name)
ValueError: Unknown hash algorithm: SHA3-224

----------------------------------------------------------------------
Ran 6714 tests in 528.063s

FAILED (errors=1)
Traceback (most recent call last):
  File "C:\Python379-32\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Python379-32\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Python379-32\lib\site-packages\Crypto\SelfTest\__main__.py", line 38,
 in <module>
    SelfTest.run(stream=sys.stdout, verbosity=1, config=config)
  File "C:\Python379-32\lib\site-packages\Crypto\SelfTest\__init__.py", line 77,
 in run
    raise SelfTestError("Self-test failed", result)
Crypto.SelfTest.SelfTestError: ('Self-test failed', <unittest.runner.TextTestRes
ult run=6714 errors=1 failures=0>)
```
I was under the impression **slow tests** were included in the standard `python -m Crypto.SelfTest` command, so what gives? 

I'm the last person to ask for an explanation, but it somehow seems to be a result of [splitting out **test vectors**](https://github.com/Legrandin/pycryptodome/commit/eda4f65718f9f9cfa3868f17675679ac583fa5bb#diff-1ec3e261925b16dc99860fe7d1f27dac79460aaf32e66dfdb22c6aae9130d3d5); when reverting back to **3.9.9** with integrated **tv**, the test succeeds: 

`python -m pip uninstall pycryptodome-test-vectors`, then 
`python -m pip uninstall pycryptodome`, then 
`python -m pip install pycryptodome==3.9.9`, then 

`python -m Crypto.SelfTest --skip-slow-tests` => 
```
Skipping slow tests
Skipping AESNI tests
Skipping test of PCLMULDQD in AES GCM
Skipping GMP tests on Windows
................................................................................
................................................................................
(snipped)
................................................................................
..................................................................
----------------------------------------------------------------------
Ran 6706 tests in 909.814s

OK
```
Thanks in advance for any insight, congrats for your overall efforts... 😄

**Comment by Legrandin** [comment]
I accidentally run into it too, and indeed it was a bug in the test suite.
The good news is that it will be fixed with https://github.com/Legrandin/pycryptodome/commit/2c22ff7b2caf3413d36060ff68996a186b6235ef .

**Comment by Legrandin** [comment]
Fixed in version v3.10.3.
